events/janus_gelfevh.c

```diff
@@ -0,0 +1,617 @@
+		if(total == 1) {
+			int n = write(sockfd, buf, len);
+			if(n < 0) {
+				JANUS_LOG(LOG_WARN, "Sending UDP message failed: %s \n", strerror(errno));
+				return -1;
+			}
+			return 1;
+		} else {
+			int offset = 0;
+			char *rnd = randstring(8);
+			for (int i = 0; i < total; i++) {
+				int bytesToSend = offset + max_gelf_msg_len < len ? max_gelf_msg_len : len - offset;
+				/* prepend the necessary headers (imitate TCP) */
+				char chunk[bytesToSend + 12];
+				chunk[0] = 0x1e;
+				chunk[1] = 0x0f;
+				memcpy(chunk + 2, rnd, 8);
+				chunk[10] = (char)i;
+				chunk[11] = (char)total;
+				char *head = chunk;
+				memcpy(head+12, buf, bytesToSend);
+				buf += bytesToSend;
+				int n = write(sockfd, head, bytesToSend + 12);
+				if(n < 0) {
+					JANUS_LOG(LOG_WARN, "Sending UDP message failed: %s \n", strerror(errno));
+					return -1;
+				}
+				offset += bytesToSend;
+				memset(chunk, 0, sizeof chunk);
+			}
+			g_free(rnd);
+		}
+	}
+	return 1;
+}
+
+int janus_gelfevh_init(const char *config_path) {
+	if(g_atomic_int_get(&stopping)) {
+		/* Still stopping from before */
+		return -1;
+	}
+	if(config_path == NULL) {
+		/* Invalid arguments */
+		return -1;
+	}
+
+	/* Read configuration */
+	gboolean enabled = FALSE;
+	char filename[255];
+	g_snprintf(filename, 255, "%s/%s.jcfg", config_path, JANUS_GELFEVH_PACKAGE);
+	JANUS_LOG(LOG_VERB, "Configuration file: %s\n", filename);
+	janus_config *config = janus_config_parse(filename);
+	if(config == NULL) {
+		JANUS_LOG(LOG_WARN, "Couldn't find .jcfg configuration file (%s), trying .cfg\n", JANUS_GELFEVH_PACKAGE);
+		g_snprintf(filename, 255, "%s/%s.cfg", config_path, JANUS_GELFEVH_PACKAGE);
+		JANUS_LOG(LOG_VERB, "Configuration file: %s\n", filename);
+		config = janus_config_parse(filename);
+	}
+	if(config != NULL) {
+		/* Handle configuration */
+		janus_config_print(config);
+		janus_config_category *config_general = janus_config_get_create(config, NULL, janus_config_type_category, "general");
+
+		/* Setup the sample event handler, if required */
+		janus_config_item *item;
+		item = janus_config_get(config, config_general, janus_config_type_item, "enabled");
+		if(!item || !item->value || !janus_is_true(item->value)) {
+			JANUS_LOG(LOG_WARN, "GELF event handler disabled (Janus API)\n");
+			goto done;
+		}
+		item = janus_config_get(config, config_general, janus_config_type_item, "backend");
+		if(!item || !item->value) {
+			JANUS_LOG(LOG_WARN, "Missing or invalid backend\n");
+			goto done;
+		}
+		backend = g_strdup(item->value);
+		item = janus_config_get(config, config_general, janus_config_type_item, "port");
+		if(!item || !item->value) {
+			JANUS_LOG(LOG_WARN, "Missing or invalid port\n");
+			goto done;
+		}
+		port = g_strdup(item->value);
+		item = janus_config_get(config, config_general, janus_config_type_item, "protocol");
+		if (item && item->value) {
+			if (strcasecmp(item->value, "udp") == 0){
+				janus_gelfevh_socket_type = 2;
+			}
```

Comment:
There should be validation here: it should make sure that it's either TCP or UDP, and other values should show a warning saying it's not an accepted value. Whether this should result in failure or fallback to a default value is up to you (I think in most plugins we show that as a warning and fall back).

Reply:
Check still missing here.

Reply:
There should be a validation here, to make sure that only "udp" or "tcp" are provided. If a different value is provided, it should either fail or show a warning and fallback to the default (udp).